block/retriever.go

```diff
@@ -116,10 +114,9 @@ func (m *Manager) ProcessNextDABatch(daMetaData *da.DASubmitMetaData) error {
 		return batchResp.Error
 	}
 
-	m.logger.Debug("retrieved batches", "n", len(batchResp.Batches), "daHeight", daMetaData.Height)
+	m.LastReceivedDAHeight.Store(daMetaData.Height)
 
-	m.retrieverMu.Lock()
```

Comment:
I think removing the retrieverMu usage here could be an issue, since i think it is used for two reasons. protecting against concurrent access to the blockCache, and concurrent block application. if we remove it here it could be possible to applyblocks here and at attemptApplyCachedBlocks at the same time. so i would keep it or i would define another mutex for block application

Reply:
agree with sergi. we need a mutex for block application.